### PR TITLE
Use type LoadBalancer instead of string

### PR DESCRIPTION
**Pull request by skmatti**
This introduces a new type for GCE Loadbalancer to replace existing string type.

/assign @MrHohn 
@cc: @freehan

### CODE REVIEW

File: pkg/utils/namer/namer.go [hunk 1]
```diff
@@ -87,6 +87,9 @@ const (
 // parameters to Namer.
 type NamerProtocol string
 
+// LoadBalancer is the name of a GCE load-balancer for an ingress.
+type LoadBalancer string
```
Comment (freehan) [inline pkg/utils/namer/namer.go]: `LoadBalancerName` ?

File: pkg/utils/namer/namer.go [hunk 2]
```diff
@@ -87,6 +87,9 @@ const (
...
+
```
Comment (freehan) [inline pkg/utils/namer/namer.go]:
```
func (n LoadBalancerName) ToString() string {
  return string(n)
}
```

File: pkg/loadbalancers/l7.go [hunk 3]
```diff
@@ -125,7 +125,7 @@ type L7 struct {
 // Warning: This should be used only for logging and should not be used to
 // retrieve/ delete gce resource names.
 func (l *L7) String() string {
-	return l.namer.LbName()
+	return string(l.namer.LoadBalancer())
```
Comment (freehan) [inline pkg/loadbalancers/l7.go]: l.namer.LoadBalancerName().ToString()

File: pkg/utils/namer/frontendnamer.go [hunk 4]
```diff
@@ -192,9 +192,9 @@ func (vn *V2IngressFrontendNamer) IsLegacySSLCert(certName string) bool {
 	return false
 }
 
-// LbName returns loadbalancer name.
+// LoadBalancer returns loadbalancer name.
 // Note that this is used for generating GCE resource names.
-func (vn *V2IngressFrontendNamer) LbName() string {
+func (vn *V2IngressFrontendNamer) LoadBalancer() LoadBalancerName {
```
Comment (freehan) [inline pkg/utils/namer/frontendnamer.go]: nit: LoadBalancerName

File: pkg/utils/namer/frontendnamer.go [hunk 5]
```diff
@@ -103,8 +103,8 @@ func (ln *V1IngressFrontendNamer) IsLegacySSLCert(certName string) bool {
 	return ln.namer.IsLegacySSLCert(ln.lbName, certName)
 }
 
-// LbName implements IngressFrontendNamer.
-func (ln *V1IngressFrontendNamer) LbName() string {
+// LoadBalancer implements IngressFrontendNamer.
+func (ln *V1IngressFrontendNamer) LoadBalancer() LoadBalancerName {
```
Comment (freehan) [inline pkg/utils/namer/frontendnamer.go]: nit: LoadBalancerName

File: pkg/utils/namer/interfaces.go [hunk 6]
```diff
@@ -33,18 +33,18 @@ type IngressFrontendNamer interface {
 	// and cert is managed by this ingress.
 	// old naming convention is of the form k8s-ssl-<lbName> or k8s-ssl-1-<lbName>.
 	IsLegacySSLCert(certName string) bool
-	// LbName returns load-balancer name for the ingress.
-	LbName() string
+	// LoadBalancer returns load-balancer name for the ingress.
+	LoadBalancer() LoadBalancerName
```
Comment (freehan) [inline pkg/utils/namer/interfaces.go]: nit: LoadBalancerName

File: pkg/utils/namer/frontendnamer.go (hunk 4 above)
Comment (skmatti) [reply pkg/utils/namer/frontendnamer.go]: Discussed offline to leave it like this, just to be consistent with other namer methods.

**Pull request merged**